src/query/helper.rs

```diff
@@ -97,6 +97,72 @@ pub trait QuerySelect: Sized {
         self
     }
 
+    /// Select columns
+    ///
+    /// ```
+    /// use sea_orm::{entity::*, query::*, tests_cfg::cake, DbBackend};
+    ///
+    /// assert_eq!(
+    ///     cake::Entity::find()
+    ///         .select_only()
+    ///         .columns([cake::Column::Id, cake::Column::Name])
+    ///         .build(DbBackend::Postgres)
+    ///         .to_string(),
+    ///     r#"SELECT "cake"."id", "cake"."name" FROM "cake""#
+    /// );
+    /// ```
+    ///
+    /// Conditionally select all columns expect a specific column
+    ///
+    /// ```
+    /// use sea_orm::{entity::*, query::*, tests_cfg::cake, DbBackend};
+    ///
+    /// assert_eq!(
+    ///     cake::Entity::find()
+    ///         .select_only()
+    ///         .columns(cake::Column::iter().filter(|col| match col {
+    ///             cake::Column::Id => false,
+    ///             _ => true,
+    ///         }))
+    ///         .build(DbBackend::Postgres)
+    ///         .to_string(),
+    ///     r#"SELECT "cake"."name" FROM "cake""#
+    /// );
+    /// ```
```

Comment:
`match` were used because neither `Column` nor `ColumnTrait` implemented `Eq`. And we cannot implement `Eq` in `Column`, it will cause nasty conflicting method name since we already have `ColumnTrait::eq` method.